packages/hardware-trezor/src/transformers/certificates.ts

```diff
@@ -0,0 +1,173 @@
+import * as Crypto from '@cardano-sdk/crypto';
+import * as Trezor from 'trezor-connect';
+import { BIP32Path } from '@cardano-sdk/crypto';
+import { Cardano } from '@cardano-sdk/core';
+import { GroupedAddress } from '@cardano-sdk/key-management';
+import { InvalidArgumentError /* , Transform*/ } from '@cardano-sdk/util';
+import { TrezorTxTransformerContext } from '../types';
+import { stakeKeyPathFromGroupedAddress } from './keyPaths';
+
+type StakeKeyCertificateType =
+  | Trezor.CardanoCertificateType.STAKE_REGISTRATION
+  | Trezor.CardanoCertificateType.STAKE_DEREGISTRATION;
+
+type TrezorStakeKeyCertificate = {
+  type: StakeKeyCertificateType;
+  path?: BIP32Path;
+  scriptHash?: Crypto.Ed25519KeyHashHex;
+  keyHash?: Crypto.Ed25519KeyHashHex;
+};
+
+type TrezorDelegationCertificate = {
+  type: Trezor.CardanoCertificateType.STAKE_DELEGATION;
+  path?: BIP32Path;
+  scriptHash?: Crypto.Ed25519KeyHashHex;
+  pool: string;
+};
+
+type TrezorPoolRegistrationCertificate = {
+  poolParameters: Trezor.CardanoPoolParameters;
+  type: Trezor.CardanoCertificateType.STAKE_POOL_REGISTRATION;
+};
+
+type ScriptHashCertCredentials = {
+  scriptHash: Crypto.Ed25519KeyHashHex;
+};
+
+type KeyHashCertCredentials = {
+  keyHash: Crypto.Ed25519KeyHashHex;
+};
+
+type PathCertCredentials = {
+  path: BIP32Path;
+};
+
+type CertCredentialsType = ScriptHashCertCredentials | KeyHashCertCredentials | PathCertCredentials;
+
+const getCertCredentials = (
+  stakeKeyHash: Crypto.Ed25519KeyHashHex,
+  knownAddresses: GroupedAddress[] | undefined
+): CertCredentialsType => {
+  const knownAddress = knownAddresses?.find(
+    (address) => Cardano.RewardAccount.toHash(address.rewardAccount) === stakeKeyHash
+  );
+  const rewardAddress = knownAddress ? Cardano.Address.fromBech32(knownAddress.rewardAccount)?.asReward() : null;
+
+  if (rewardAddress?.getPaymentCredential().type === Cardano.CredentialType.KeyHash) {
+    const path = stakeKeyPathFromGroupedAddress(knownAddress);
```

Comment:
Im getting the following eslint error here:
<img width="1508" alt="grafik" src="https://github.com/input-output-hk/cardano-js-sdk/assets/172414/ef68c13a-35a6-4a43-b73c-e91202c97787">

Reply:
@DominikGuzei everything is green on my side and also the pre-commit hook didn't catch it.
Also, [stakeKeyPathFromGroupedAddress](https://github.com/input-output-hk/cardano-js-sdk/blob/chore/lw-6800-certificates-and-aux-mapper-for-trezor-sdk-implementation/packages/hardware-trezor/src/transformers/keyPaths.ts#L19) is accepting `GroupedAddress | undefined`

Reply:
You probably merged this branch into your miscellaneous mapper. Did you maybe change something in https://github.com/input-output-hk/cardano-js-sdk/blob/chore/lw-6800-certificates-and-aux-mapper-for-trezor-sdk-implementation/packages/hardware-trezor/src/transformers/keyPaths.ts?
Also, can you check it in this PR `chore/lw-6800-certificates-and-aux-mapper-for-trezor-sdk-implementation` branch ?

Reply:
Hm interesting, I didn't change anything outside `tx.ts` 🤔

Reply:
@tomislavhoracek all good, it was actually a caching issues of Webstorm 🙈 when i followed the type signature of the function it updated and the error disappeared 👍

Reply:
Thanks D ❤️